src/GafferBindings/CompoundNumericPlugBinding.cpp

```diff
@@ -55,58 +55,59 @@ using namespace Gaffer;
 namespace
 {
 
-template<typename T>
-std::string compoundNumericPlugRepr( const T *plug )
+std::string maskedCompoundNumericPlugRepr( const ValuePlug *plug, unsigned flagsMask )
 {
-	return maskedCompoundNumericPlugRepr( plug, Plug::All );
-}
+	std::string extraArgs = "";
 
-template<typename T>
-class CompoundNumericPlugSerialiser : public ValuePlugSerialiser
-{
-  public:
-
-	static std::string repr( const Gaffer::ValuePlug *plug, unsigned flagsMask, const std::string &extraArguments, const Serialisation *serialisation )
+	if( const V3fPlug *v3fPlug = IECore::runTimeCast<const V3fPlug>( plug ) ) 
 	{
-		std::string result = ValuePlugSerialiser::repr( plug, flagsMask, extraArguments, serialisation );
-		if( const V3fPlug *v3fPlug = static_cast<const V3fPlug *>( plug ))
+		// TODO: we currently can't do this, but that would remove some of the uglyness below
+		// std::string extraArgs = IECorePython::repr( interpretation );
+
+		const IECore::GeometricData::Interpretation interpretation = v3fPlug->interpretation(); 
+		extraArgs = "interpretation = IECore.GeometricData.Interpretation.";
+
+		switch( interpretation )
 		{
-			std::string interpretation;
-			switch( v3fPlug->getInterpretation() )
-			{
-			case IECore::GeometricData::Point :
-			{
-				interpretation = "Point";
-				break;
-			}
 			case IECore::GeometricData::Normal :
 			{
-				interpretation = "Normal";
+				extraArgs += "Normal";
 				break;
 			}
-			case IECore::GeometricData::Color :
+			case IECore::GeometricData::Point :
 			{
-				interpretation = "Color";
+				extraArgs += "Point";
 				break;
 			}
 			case IECore::GeometricData::Vector :
 			{
-				interpretation = "Vector";
+				extraArgs += "Vector";
 				break;
 			}
-			default :
-				interpretation = "Numeric";
+			default:
+				extraArgs += "None";
 				break;
-			}
-
-			boost::replace_first( result, "(", boost::str( boost::format( "( Gaffer._Gaffer.Interpretation.%s, ") % interpretation ));
 		}
-		return result;
 	}
 
+	return ValuePlugSerialiser::repr( plug, flagsMask, extraArgs );
+}
+
+template<typename T>
+std::string compoundNumericPlugRepr( const T *plug )
+{
+	return maskedCompoundNumericPlugRepr( plug, Plug::All );
+}
+
+template<typename T>
+class CompoundNumericPlugSerialiser : public ValuePlugSerialiser
+{
+  public:
+
 	virtual std::string constructor( const Gaffer::GraphComponent *graphComponent, const Serialisation &serialisation ) const
 	{
-		return repr( static_cast<const ValuePlug *>( graphComponent ), Plug::All & ~Plug::ReadOnly, "", &serialisation );
+		printf("constructor() called...\n");
+		return maskedCompoundNumericPlugRepr( static_cast<const ValuePlug *>( graphComponent ), Plug::All & ~Plug::ReadOnly );
```

Comment:
This cast isn't right - `maskedCompoundNumericPlugRepr()` demands a CompoundNumericPlug of some type (once my suggested changes are made), so we should pass `static_cast<const T *>( graphComponent )`. The cast is safe because serialisers are only called with the types for which they are registered, and we register `CompoundNumericPlugSerialiser<T>` for type `T`.